guard/src/main.rs

```diff
@@ -1,87 +1,41 @@
-        .arg_required_else_help(true);
+    let args = CfnGuard::parse();
 
-    let mut commands = utils::get_guard_commands();
-    commands.push(Box::<commands::completions::Completions>::default());
-
-    let mappings = commands.iter().map(|s| (s.name(), s)).fold(
-        HashMap::with_capacity(commands.len()),
-        |mut map, entry| {
-            map.insert(entry.0, entry.1.as_ref());
-            map
+    let mut writer = match &args.command {
+        Commands::ParseTree(cmd) => match &cmd.output {
+            Some(path) => Writer::new(WBFile(File::create(path)?), Stderr(std::io::stderr())),
+            None => Writer::new(Stdout(std::io::stdout()), Stderr(std::io::stderr())),
         },
-    );
-
-    for each in &commands {
-        app = app.subcommand(each.command());
-    }
-
-    let help = app.render_usage();
-    let app = app.get_matches();
+        Commands::Rulegen(cmd) => match &cmd.output {
+            Some(path) => Writer::new(WBFile(File::create(path)?), Stderr(std::io::stderr())),
+            None => Writer::new(Stdout(std::io::stdout()), Stderr(std::io::stderr())),
+        },
+        _ => Writer::new(Stdout(std::io::stdout()), Stderr(std::io::stderr())),
+    };
 
-    match app.subcommand() {
-        Some((name, value)) => {
-            if let Some(command) = mappings.get(name) {
-                let mut output_writer: Writer = if [PARSE_TREE, RULEGEN].contains(&command.name()) {
-                    let writer: Writer = match value.get_one::<String>(OUTPUT.0) {
-                        Some(file) => {
-                            Writer::new(WBFile(File::create(file)?), Stderr(std::io::stderr()))
-                        }
-                        None => Writer::new(Stdout(std::io::stdout()), Stderr(std::io::stderr())),
-                    };
-                    writer
-                } else {
-                    Writer::new(Stdout(std::io::stdout()), Stderr(std::io::stderr()))
-                };
+    let mut reader = Reader::new(ReadBuffer::Stdin(std::io::stdin()));
 
-                match (*command).execute(
-                    value,
-                    &mut output_writer,
-                    &mut Reader::new(ReadBuffer::Stdin(std::io::stdin())),
-                ) {
-                    Err(e) => {
-                        output_writer
-                            .write_err(format!("Error occurred {e}"))
-                            .expect("failed to write to stderr");
+    match args.execute(&mut writer, &mut reader) {
+        Ok(code) => exit(code),
+        Err(e) => {
+            writer
+                .write_err(format!("Error occurred {e}"))
+                .expect("failed to write to stderr");
 
-                        exit(-1);
-                    }
-                    Ok(code) => exit(code),
-                }
-            } else {
-                println!("{}", help);
-            }
-        }
-        None => {
-            println!("{}", help);
+            exit(-1)
```

Comment:
You didn't do this now but just curious - why -1 vs 1?

Reply:
I'm not sure for the reasoning behind this, currently we use exit code of 1 to denote an error that we caught during our test command, so that is why we cannot currently use an exit code of 1 over here. These exit codes were decided upon before I joined the project though.